src/Components/WishlistProductCard/WishlistProductCard.jsx

```diff
@@ -0,0 +1,237 @@
+    const [wishlistHeartIcon, setWishlistHeartIcon] = useState("fa-heart-o")
+    const [wishlistBtn, setWishlistBtn]             = useState("add-to-wishlist-btn")
+
+    useEffect(()=>{
+        const index = userWishlist.findIndex(product=> {
+            return product._id === productdetails._id
+        })
+
+        if(index!==-1)
+        {
+            setWishlistHeartIcon("fa-heart")
+            setWishlistBtn("added-to-wishlist-btn")
+        }
+        else
+        {
+            setWishlistHeartIcon("fa-heart-o")
+            setWishlistBtn("add-to-wishlist-btn")
+        }
+    },[userWishlist, productdetails._id, setWishlistHeartIcon, setWishlistBtn])
+
+    async function addOrRemoveItemToWishlist()
+    {
+        if(wishlistHeartIcon==="fa-heart-o" && wishlistBtn ==="add-to-wishlist-btn")
+        {
+            //Item not present in wishlist, add it
+            const token=localStorage.getItem('token')
+
+            if(token)
+            {
+                const user = jwt_decode(token)
+                
+                if(!user)
+                {
+                    localStorage.removeItem('token')
+                    showToast("warning","","Kindly Login")
+                    navigate('/login')
+                }
+                else
+                {
+                    let wishlistUpdateResponse = await axios.patch(
+                        "https://bookztron.herokuapp.com/api/wishlist",
+                        {
+                            productdetails
+                        },
+                        {
+                            headers:
+                            {
+                                'x-access-token': localStorage.getItem('token'),
+                            }
+                        }
+                    )
+            
+                    if(wishlistUpdateResponse.data.status==="ok")
+                    {
+                        setWishlistHeartIcon("fa-heart")
+                        setWishlistBtn("added-to-wishlist-btn")
+                        dispatchUserWishlist({type: "UPDATE_USER_WISHLIST",payload: wishlistUpdateResponse.data.user.wishlist})
+                        showToast("success","","Item successfully added to wishlist")
+                    }
+                }
+            }
+            else
+            {
+                showToast("warning","","Kindly Login")
+            }   
+        }
+        else
+        {
+            //Item present in wishlist, remove it
+            const token=localStorage.getItem('token')
+
+            if(token)
+            {
+                const user = jwt_decode(token)
+                
+                if(!user)
+                {
+                    localStorage.removeItem('token')
+                    showToast("warning","","Kindly Login")
+                    navigate('/login')
+                }
+                else
+                {
+                    let wishlistUpdateResponse = await axios.delete(
```

Comment:
create a sperate folder , to hit all backend request 

it help to sport backend response & return response to all component when ever it needed